crates/evm/src/execute.rs

```diff
@@ -499,6 +502,55 @@ where
     }
 }
 
+/// Creates an `EvmState` from a map of balance increments and the current state
+/// to load accounts from.
+/// Zero balance increments are ignored and won't create state entries.
+pub fn balance_increment_state<DB>(
+    balance_increments: &HashMap<Address, u128, DefaultHashBuilder>,
+    state: &mut State<DB>,
+) -> Result<EvmState, BlockExecutionError>
+where
+    DB: Database,
+{
+    let mut create_account =
+        |address: &Address, balance_inc: u128| -> Result<(Address, Account), BlockExecutionError> {
+            let account = state.load_cache_account(*address).map_err(|_| {
+                BlockExecutionError::msg("could not load account for balance increment")
+            })?;
+
+            let transition_account = account
+                .increment_balance(balance_inc)
+                .ok_or(BlockValidationError::IncrementBalanceFailed)?;
```

Comment:
balance was already incremented prior to this call. this does it 2nd time

Reply:
@rkrasiuk very good catch thx! should be fixed in https://github.com/paradigmxyz/reth/pull/13050/commits/cc7bd7632ba8b6e78f26478744a2702cebc44b48 also added a test to cover the issue, ptal